### PR TITLE
[tests-only][full-ci] add tests to check activities after public actions

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds tests to check api activities of public user's actions (add|edit|delete) on a publicly shared resource.
``` feature
Scenario: user checks public's activities on a publicly shared file
Scenario: user checks public's activities on a publicly shared folder
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/9543
- Bug Report: https://github.com/owncloud/ocis/issues/9676

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 1]
```diff
@@ -1104,3 +1104,467 @@ Feature: check share activity
       | permissionsRole | Viewer       |
     When user "Brian" tries to list the activities of file "anotherTextfile.txt" from space "Personal" owned by user "Alice" using the Graph API
     Then the HTTP status code should be "403"
+
+  @issue-9676
+  Scenario: user checks public's activities on a publicly shared file
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1109]:
```suggestion
  Scenario: user checks public activities of a link shared file
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 2]
```diff
@@ -1104,3 +1104,467 @@ Feature: check share activity
...
+                }
+              ]
+            }
+          }
+        }
+      }
+      """
+
+  @issue-9676
+  Scenario: user checks public's activities on a publicly shared folder
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1248]:
```suggestion
  Scenario: user checks public activities of a link shared folder
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 3]
```diff
@@ -1104,3 +1104,467 @@ Feature: check share activity
...
+                                }
+                              }
+                            },
+                            "user": {
+                              "type": "object",
+                              "required": ["id","displayName"],
+                              "properties": {
+                                "id": {
+                                  "type": "string",
+                                  "pattern": "[a-zA-z]{15}"
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1217]: what is this id pattern? user-id?

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 4]
```diff
@@ -154,6 +154,23 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
 		);
 	}
 
+	/**
+	 * @Given /^the public has deleted (?:file|folder|entry) "([^"]*)" from the last public link share using the password "([^"]*)" and (old|new) public WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:158]:
```suggestion
	 * @Given /^the public has deleted (?:file|folder|entry) "([^"]*)" from the last link share with password "([^"]*)" using the public WebDAV API$/
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 5]
```diff
@@ -554,6 +571,26 @@ public function publiclyUploadingContentWithPassword(
 		);
 	}
 
+	/**
+	 * @Given /^the public has uploaded file "([^"]*)" with password "([^"]*)" and content "([^"]*)" using the (old|new) public WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:575]:
```suggestion
	 * @Given /^the public has uploaded file "([^"]*)" with content "([^"]*)" and password "([^"]*)" to the last link share using the public WebDAV API$/
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 3 above)
Comment (prashant-gurung899) [reply tests/acceptance/features/apiActivities/shareActivities.feature:1217]: @saw-jan 
The response looks like following :
```
"user": {
"id": "gDtQjXsVvHqForR",
"displayName": "Public"
}
```

So, I used `"[a-zA-z]{15}"`  regex to match it.

Comment (saw-jan) [reply tests/acceptance/features/apiActivities/shareActivities.feature:1217]: it is is link share token

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 6]
```diff
@@ -155,20 +154,29 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
...
+
+	/**
+	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last public link share using the password "([^"]*)" using the public WebDAV API$/
+	 *
+	 * @param string $file
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function thePublicDeletesFileFromTheLastPublicShareUsingThePasswordPasswordAndOldPublicWebdavApi(string $file, string $password):void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:177]:
```suggestion
	public function thePublicDeletesFileFromTheLastPublicShareUsingThePasswordPasswordAndOldPublicWebdavApi(string $file, string $password): void {
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 7]
```diff
@@ -155,20 +154,29 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
 	}
 
 	/**
-	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last public link share using the password "([^"]*)" and (old|new) public WebDAV API$/
+	 * @Given /^the public has deleted (?:file|folder|entry) "([^"]*)" from the last public link share with password "([^"]*)" using the public WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:157]:
```suggestion
	 * @Given /^the public has deleted (?:file|folder|entry) "([^"]*)" from the last link share with password "([^"]*)" using the public WebDAV API$/
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 8]
```diff
@@ -155,20 +154,29 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
...
 	 * @param string $password
-	 * @param string $publicWebDAVAPIVersion
 	 *
 	 * @return void
 	 */
-	public function thePublicDeletesFileFromTheLastPublicShareUsingThePasswordPasswordAndOldPublicWebdavApi(string $file, string $password, string $publicWebDAVAPIVersion):void {
-		if ($publicWebDAVAPIVersion === "old") {
-			return;
-		}
+	public function thePublicHasDeletedFileFromTheLastPublicLinkShareUsingThePasswordAndNewPublicWebdavApi(string $file, string $password): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:164]:
```suggestion
	public function thePublicHasDeletedFileFromTheLastLinkShareWithPasswordUsingPublicWebdavApi(string $file, string $password): void {
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 9]
```diff
@@ -155,20 +154,29 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
...
+		$response = $this->deleteFileFromPublicShare($file, $password);
+		$this->featureContext->theHTTPStatusCodeShouldBe([201, 204], "", $response);
+	}
+
+	/**
+	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last public link share using the password "([^"]*)" using the public WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:170]:
```suggestion
	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last link share using the password "([^"]*)" using the public WebDAV API$/
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php (hunk 6 above)
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:177]:
```suggestion
	public function thePublicDeletesFileFromTheLastShareWithPasswordUsingPublicWebdavApi(string $file, string $password):void {
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 10]
```diff
@@ -534,26 +542,41 @@ public function thePublicHasUploadedFileWithContentWithAutoRenameMode(string $fi
...
+	/**
+	 * @Given /^the public has uploaded file "([^"]*)" with content "([^"]*)" and password "([^"]*)" to the last link share using the public WebDAV API$/
+	 *
+	 * @param string $filename
+	 * @param string $content
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function thePublicHasUploadedFileWithPasswordAndContentUsingTheNewPublicWebdavApi(string $filename, string $content = 'test', string $password = ''): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:571]:
```suggestion
	public function thePublicHasUploadedFileWithContentAndPasswordToLastLinkShareUsingPublicWebdavApi(string $filename, string $content = 'test', string $password = ''): void {
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 11]
```diff
@@ -1104,3 +1104,467 @@ Feature: check share activity
...
+
+  @issue-9676
+  Scenario: user checks public activities of a link shared folder
+    Given using SharingNG
+    And user "Alice" has created folder "/FOLDER"
+    And user "Alice" has created the following resource link share:
+      | resource        | /FOLDER   |
+      | space           | Personal |
+      | permissionsRole | edit     |
+      | password        | %public% |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1255]:
```suggestion
      | resource        | /FOLDER  |
      | space           | Personal |
      | permissionsRole | edit     |
      | password        | %public% |
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 12]
```diff
@@ -155,20 +154,29 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
...
+
+	/**
+	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last link share using the password "([^"]*)" using the public WebDAV API$/
+	 *
+	 * @param string $file
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function thePublicDeletesFileFromTheLastShareWithPasswordUsingPublicWebdavApi(string $file, string $password): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:177]:
```suggestion
	public function thePublicDeletesFileFromTheLastLinkShareWithPasswordUsingPublicWebdavApi(string $file, string $password): void {
```

File: tests/acceptance/bootstrap/PublicWebDavContext.php [hunk 13]
```diff
@@ -155,20 +154,29 @@ public function deleteFileFromPublicShare(string $fileName, string $publicWebDAV
...
-		if ($publicWebDAVAPIVersion === "old") {
-			return;
-		}
+	public function thePublicHasDeletedFileFromTheLastLinkShareWithPasswordUsingPublicWebdavApi(string $file, string $password): void {
+		$response = $this->deleteFileFromPublicShare($file, $password);
+		$this->featureContext->theHTTPStatusCodeShouldBe([201, 204], "", $response);
+	}
+
+	/**
+	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last link share using the password "([^"]*)" using the public WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/PublicWebDavContext.php:170]:
```suggestion
	 * @When /^the public deletes (?:file|folder|entry) "([^"]*)" from the last link share with password "([^"]*)" using the public WebDAV API$/
```

**Review by saw-jan — APPROVED** [review]
:+1:

**Review by amrita-shrestha — APPROVED** [review]

**Pull request merged**